plugins/language_nim.lua

```diff
@@ -88,14 +88,37 @@ for _, type in ipairs(standard_generic_types) do
   table.insert(patterns, { pattern = type.." +%f[%w]", type = "keyword2" })
 end
 
+local function string_pattern(start, stop, syntax)
+  return {
+    pattern = { start, stop, '\\' },
+    type    = "string",
+    syntax  = syntax and {
+      patterns = syntax,
+      symbols  = {},
+    } or nil
+  }
+end
+
+local interpolation_syntax = {
+  { pattern = { '{{', '}}' }, type = "string" },
+  { pattern = { '{', '}', '\\' }, type = "keyword2", syntax = { patterns = patterns, symbols = symbols, } },
+  { pattern = "[%S][%w]*", type = "string" },
+}
+
 local user_patterns = {
   -- comments
   { pattern = { "##?%[", "]##?" },            type = "comment" },
   { pattern = "##?.-\n",                      type = "comment" },
   -- strings and chars
-  { pattern = { '"', '"', '\\' },             type = "string" },
-  { pattern = { '"""', '"""[^"]' },           type = "string" },
   { pattern = { "'", "'", '\\' },             type = "literal" },
+  string_pattern('"""', '"""[^"]'),
+  string_pattern('"'  , '"'      ),
+  string_pattern('\\"', '\\"'    ), -- For highlighting strings inside iterpolated blocks
+  -- string interpolation
+  string_pattern('%&"""' , '"""[^"]', interpolation_syntax),
+  string_pattern('fmt"""', '"""[^"]', interpolation_syntax),
```

Comment:
`[^"]` is a bit problematic, as it captures the character.
Look into either using Lua frontier patterns or regex with negative lookaheads.

Reply:
Ok, I'll have a look. I just kept it from the original syntax

Reply:
I made the change for frontier matches as I understand them and have the same result as in my original image